cpp/cmake_modules/BuildUtils.cmake

```diff
@@ -173,21 +191,26 @@ function(ADD_ARROW_LIB LIB_NAME)
     # because of dllexport declarations on Windows.
     # The Xcode generator doesn't reliably work with Xcode as target names are not
     # guessed correctly.
-    set(LIB_DEPS ${ARG_SOURCES})
-    set(EXTRA_DEPS ${ARG_DEPENDENCIES})
-
-    if(ARG_EXTRA_INCLUDES)
-      set(LIB_INCLUDES ${ARG_EXTRA_INCLUDES})
-    endif()
+    set(USE_OBJLIB OFF)
   else()
-    # Otherwise, generate a single "objlib" from all C++ modules and link
+    set(USE_OBJLIB ON)
+  endif()
+
+  if(USE_OBJLIB)
+    # Generate a single "objlib" from all C++ modules and link
     # that "objlib" into each library kind, to avoid compiling twice
     add_library(${LIB_NAME}_objlib OBJECT ${ARG_SOURCES})
     # Necessary to make static linking into other shared libraries work properly
     set_property(TARGET ${LIB_NAME}_objlib PROPERTY POSITION_INDEPENDENT_CODE 1)
     if(ARG_DEPENDENCIES)
       add_dependencies(${LIB_NAME}_objlib ${ARG_DEPENDENCIES})
     endif()
+    if(ARG_PRECOMPILED_HEADER_LIB AND ARROW_USE_PRECOMPILED_HEADERS)
```

Comment:
Could you remove `AND ARROW_USE_PRECOMPILED_HEADERS` because this is checked in `reuse_precompiled_header_lib`?

Reply:
Will do.